skimage/exposure/_adapthist.py

```diff
@@ -13,21 +13,22 @@
 responsible.  Basically, don't be a jerk, and remember that anything free
 comes with no guarantee.
 """
+from __future__ import division
+import numbers
 import numpy as np
 from .. import img_as_float, img_as_uint
 from ..color.adapt_rgb import adapt_rgb, hsv_value
 from ..exposure import rescale_intensity
-from ..util import view_as_blocks, pad
+from ..util import view_as_windows
+from .._shared.utils import skimage_deprecation
 
 
-MAX_REG_X = 16  # max. # contextual regions in x-direction */
-MAX_REG_Y = 16  # max. # contextual regions in y-direction */
 NR_OF_GREY = 2 ** 14  # number of grayscale levels to use in CLAHE algorithm
 
 
 @adapt_rgb(hsv_value)
-def equalize_adapthist(image, ntiles_x=8, ntiles_y=8, clip_limit=0.01,
-                       nbins=256):
+def equalize_adapthist(image, kernel_size=64, ntiles_x=None, ntiles_y=None,
+                       clip_limit=0.01, nbins=256):
```

Comment:
@jni the results aren't guaranteed to be equivalent because the previous behavior acted badly at the borders in some (not all) cases.

Order of kwargs is a tricky one, though. From an API perspective, if people called `equalize_adapthist(im, 8, 8, 0.001, 128)` previously you are correct - the only way not to break that API is dumping the new kwarg at the end. Another good reason to name your kwargs ;) Still, Python allows this use mode so someone may be using it.

Unfortunately, as the kernel size is crucial to the adaptive nature of this algorithm, control over this _should_ immediately follow the primary argument. So we either hold fast to our API breakage policy and ship a poor API as a result, or we break it.

In this specific case, I'd recommend breaking the API in the name of having a good future API path. We could make the change slightly less disruptive by ordering the arguments `image, ntiles_x, ntiles_y, kernel_size, ...` but I don't think having `kernel_size` at the end is a great move. And, while it is technically a "breakage" of the API, it will affect only a subset of users.